Remove qt4 sans

DavidFair opened the pull request:
**Description of work.**

Removes MantidPlot imports and calls from the SANS codebase
In the majority of cases I've replaced them with NotImplementedErrors either because the module is deprecated, or because it's been broken since Mantid 6.0 without users having noticed it.

The intention is that instead of silently not plotting anything we explicitly tell users it's not working with workbench please contact us if you need this.

**To test:**
- Download ISIS Training data
- Ensure LOQDemo folder is present
- Open ISIS SANS interface
- Load a .txt or .toml file from the LOQ Demo folder
- Enter 74044 into the left hand column
- Hit load
- Go to Settings, Mask, ensure row 0 is selected (hit update rows if not)
- Display mask should open the instrument viewer

There is no associated issue. - This was noticed whilst working on plotting code

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DavidFair committed:
Remove Mantidplot from ISIS SANS GUI presenters

Removes Mantid Plot handling code from SANS GUI presenters as part of
maintenance

DavidFair committed:
Remove MantidPlot calls from deprecated ISISCommandInterface

Removes Mantidplot calls from deprecated interface, instead replacing it
with NotImplementedErrors. This encourages users to move their code that
plots over to the new interface, or set plot to false without us having
to continue to support it onwards

DavidFair committed:
Remove Mantidplot parts from support command interface

Add a not implemented error, since this has not been raised yet and we
want users to contact us if it's a problem for their scripts they are
writing.

As this would have been broken for at least 2 releases without being
noticed, we can presume this isn't critical to get replaced with a
workbench equiv

gemmaguest reviewed (changes requested):
Code changes look good and works as expected :+1: Just a couple of very minor suggestions

DavidFair committed:
Improve docs string and error message

Removes a typo from docs string
Changes error message to point users to forum, instead of generically
contacting us

gemmaguest commented on scripts/SANS/sans/command_interface/ISISCommandInterface.py:
```diff
@@ -1020,8 +1013,8 @@ def PhiRanges(phis, plot=True):
-    if plot and mantidplot:
-        mantidplot.plotSpectrum(reduced_workspace_names, 0)
+    if plot:
+        raise NotImplementedError("Plotting is not implemented for workbench yet, please contact the dev team.")
```
Same here regarding the forum

gemmaguest commented on scripts/SANS/sans/command_interface/ISISCommandInterface.py:
```diff
@@ -1076,6 +1069,8 @@ def PlotResult(workspace, canvas=None):
         @param canvas: optional handle to an existing graph to write the plot to
         @return: a handle to the graph that was written to
     """
+    raise NotImplementedError("Plotting is not implemented for workbench yet, please contact the dev team.")
```
Same here regarding the forum

DavidFair replied:
The worst part is I thought, "I'm sure I've written this more than once", then promptly didn't check :face_with_head_bandage:

gemmaguest reviewed (changes requested):
Just a couple more places we should mention the forum!
